pkg/reconciler/trigger/controller.go

```diff
@@ -58,11 +58,11 @@ const (
 var filterBroker = pkgreconciler.AnnotationFilterFunc(eventingv1beta1.BrokerClassAnnotationKey, brokerv1beta1.BrokerClass, false /*allowUnset*/)
 
 func NewController(ctx context.Context, cmw configmap.Watcher) *controller.Impl {
-	// TODO initialize project ID here via env var or metadata.
-	projectID := ""
-
 	triggerInformer := triggerinformer.Get(ctx)
 
+	// If there is an error, the projectID will be empty. The reconciler will retry
+	// to get the projectID during reconciliation.
+	projectID, _ := utils.ProjectID(os.Getenv(utils.ProjectIDEnvKey), metadataClient.NewDefaultMetadataClient())
 	// Attempt to create a pubsub client for all worker threads to use. If this
 	// fails, pass a nil value to the Reconciler. They will attempt to
 	// create a client on reconcile.
```

Comment:
Why is Trigger reconciler still using the old init style? Is this planned to be changed in a future PR?

Reply:
Which old init style exactly? I think both broker and trigger controllers are using the same init style.

Reply:
Disregard this, I thought `newPubsubClient` was not used in broker/controller.go anymore, but it is back now (or maybe never left).